### PR TITLE
requireNewlineAfterArrayElements: impl

### LINKED ISSUE #317: New rule: validateNewlineAfterArrayElements to force each item to be on a new line
An option something like = `requireNewLinesInArray = true;`

Invalid:
```js
var fruit = [ 'apples', 'bananas', 'pears'];
```

Valid:
```js
var fruit = [ 
  'apples', 
  'bananas', 
  'pears'
];
```

There might need to be other options, such as `firstItemOnNewLine`.

Some advantages for this style is that it makes it easier to follow diffs and it is easier to read when there are many items.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1996091-new-rule-requirenewlineafterarrayelements-to-force-each-item-to-be-on-a-new-line?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by zxqfox**
Fixes #317

### CODE REVIEW

**Comment by zxqfox** [comment]
Rebased, checked coverage.

File: README.md [hunk 1]
```diff
@@ -3400,6 +3400,48 @@ while (cond){
...
+Type: `Boolean` or `Number` (maximum) or `Object` (maximum: Infinity, ignoreBrackets: false)
+
+Values: `true`
+
+Default: Infinity
+
+#### Example
+
+```js
+"requireNewlineAfterArrayElement": 3
```
Comment (markelog) [inline README.md]: For the description of the rule i'm not sure what id does, could add more clarify it more explicitly?

Comment (zxqfox) [inline README.md]: id?

"Requires each element in array on a new line when size is more than passed number."

"Requires each element in array on a single line when length is bigger than `maximum`."

hm?

**Comment by zxqfox** [comment]
Rebased, moved rule description to `js` and improved it.

@markelog Please check this out. My english isn't perfect.

**Comment by zxqfox** [comment]
Oh, looks like we forgot about this.

**Comment by zxqfox** [comment]
@markelog Can you take a final look here?

**Comment by markelog** [comment]
We have a falling CI-run, also, since there couldn't be a `disallow` rule, perhaps we can call it `validateNewlineAfterArrayElements`, maybe even use it as general rule for naming - if there couldn't be an opposite rule, start name from "validate"

**Comment by zxqfox** [comment]
> We have a falling CI-run

It's a "Line must be at most 120 characters at lib/rules/require-newline-after-array-elements.js"

> maybe even use it as general rule for naming - if there couldn't be an opposite rule, start name from "validate"

Agreed. Let's do this!

**Comment by markelog** [comment]
Oh, and nitpick: be sure to use imperative mood in commit head

**Comment by zxqfox** [comment]
@markelog Oh, lol. Thanks!

**Pull request closed without merging**